pkg/operator/controller/controller_router_deployment.go

```diff
@@ -147,17 +147,21 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, routerImage strin
 		}
 	}
 
+	nodeSelector := map[string]string{
+		"beta.kubernetes.io/os":          "linux",
+		"node-role.kubernetes.io/worker": "",
+	}
 	if ci.Spec.NodePlacement != nil {
 		if ci.Spec.NodePlacement.NodeSelector != nil {
-			nodeSelector, err := metav1.LabelSelectorAsMap(ci.Spec.NodePlacement.NodeSelector)
+			var err error
+			nodeSelector, err = metav1.LabelSelectorAsMap(ci.Spec.NodePlacement.NodeSelector)
 			if err != nil {
 				return nil, fmt.Errorf("clusteringress %q has invalid spec.nodePlacement.nodeSelector: %v",
 					ci.Name, err)
 			}
-
-			deployment.Spec.Template.Spec.NodeSelector = nodeSelector
 		}
 	}
+	deployment.Spec.Template.Spec.NodeSelector = nodeSelector
```

Comment:
Just a random thought... I wonder how changing this affects support commitments. For example, we don't support scheduling the ingresscontrollers on masters.

Reply:
> Just a random thought... I wonder how changing this affects support commitments. For example, we don't support scheduling the ingresscontrollers on masters.

This reminds me of our earlier merge-versus-replace discussion.  Should the API even allow disabling the `node-role.kubernetes.io/worker` selector if an ingresscontroller without it is unsupported?